parsl/executors/threads.py

```diff
@@ -21,7 +20,7 @@ class ThreadPoolExecutor(ParslExecutor, RepresentationMixin):
         Number of threads. Default is 2.
     thread_name_prefix : string
         Thread name prefix (only supported in python v3.6+).
-    storage_access : list of :class:`~parsl.data_provider.scheme.Scheme`
+    storage_access : list of :class:`~parsl.data_provider.staging.Staging`
```

Comment:
I think we need to flesh out the docstring a bit to help people that aren't reading the source. What is the default? What happens if you pass `None`? What happens if you pass `[]`?

Also, consider copy-pasting this to the other executors, which don't appear to have any entries for `storage_access`. (I see this problem existed before your changes, but this seems like as good a time as any to add it. I'm also happy to defer this to a 'make it possible for users to understand data management by updating all docs and docstrings' issue.)

Reply:
I would like to write some user-guide chapter on "What is file handling/staging" which this can refer to. Not in this PR, though.

Reply:
:+1: